src/figuresClasses.ts

```diff
@@ -2,10 +1,72 @@
+export interface Figure {
+  shape: 'triangle' | 'rectangle' | 'circle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('not triangle');
+    } else if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('not triangle');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Number(area.toFixed(2));
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: Figure['shape'] = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Figure['color'],
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The condition should use `this.radius` instead of `radius` to correctly reference the instance variable.